amti/actions/create.py

```diff
@@ -140,6 +140,71 @@ def initialize_batch_directory(
     return batch_dir
 
 
+def estimate_batch_cost(batch_dir):
+    """
+    Estimate the cost of the given batch.
+
+    Parameters
+    ----------
+    batch_dir : str
+        the path to the batch directory.
+
+    Returns
+    -------
+    float
+        the estimated cost of uploading this batch to MTURK (in USD),
+        including MTURK overhead.
+    """
+    # construct all necessary paths
+    _, batch_dir_subpaths = settings.BATCH_DIR_STRUCTURE
+    definition_dir_name, definition_dir_subpaths = \
+        batch_dir_subpaths['definition']
+    hittype_properties_file_name, _ = \
+        definition_dir_subpaths['hittype_properties']
+    hit_properties_file_name, _ = definition_dir_subpaths['hit_properties']
+    data_file_name, _ = batch_dir_subpaths['data']
+
+    hittype_properties_path = os.path.join(
+        batch_dir,
+        definition_dir_name,
+        hittype_properties_file_name)
+    hit_properties_path = os.path.join(
+        batch_dir,
+        definition_dir_name,
+        hit_properties_file_name)
+    hittype_properties_path = os.path.join(
+        batch_dir,
+        definition_dir_name,
+        hittype_properties_file_name)
+    hit_properties_path = os.path.join(
+        batch_dir,
+        definition_dir_name,
+        hit_properties_file_name)
+    data_path = os.path.join(
+        batch_dir, data_file_name)
+
+    # Load relevant files
+
+    with open(hittype_properties_path, 'r') as hittype_properties_file:
+        hittype_properties = json.load(hittype_properties_file)
+
+    with open(hit_properties_path, 'r') as hit_properties_file:
+        hit_properties = json.load(hit_properties_file)
+
+    with open(data_path, "r") as data_file:
+        num_of_hits = len([line for line in data_file
+                           if line.strip()])
```

Comment:
Changing this line to:

```python
num_of_hits = sum(1 for _ in data_file)
```

counts the lines without explicitly creating a list in memory.